db/index.go

```diff
@@ -123,16 +123,24 @@ func (i *collectionBaseIndex) getDocFieldValue(doc *client.Document) ([][]byte,
 	result := make([][]byte, 0, len(i.fieldsDescs))
 	for iter := range i.fieldsDescs {
 		fieldVal, err := doc.GetValue(i.fieldsDescs[iter].Name)
+		isNil := false
 		if err != nil {
 			if errors.Is(err, client.ErrFieldNotExist) {
-				valBytes, err := client.NewFieldValue(client.LWW_REGISTER, nil).Bytes()
-				if err != nil {
-					return nil, err
-				}
-				result = append(result, valBytes)
-				continue
+				isNil = true
+			} else {
+				return nil, err
 			}
-			return nil, err
+		}
+		if fieldVal != nil && fieldVal.Value() == nil {
+			isNil = true
+		}
+		if isNil {
+			valBytes, err := client.NewFieldValue(client.LWW_REGISTER, nil).Bytes()
```

Comment:
suggestion: This does look a bit strange, can we perhaps instead have a const in the client package for this:

```
// With documentation and probably a better name
const NIL_FIELD_BYTES []byte = ...
```

Reply:
this will be gone as well with new encoding. I can add a comment though.

Reply:
Nice - thanks Islam, same as comment on `if len(key.FieldValues[i]) == 1 && key.FieldValues[i][0] == cborNil` - I'm happy as-is if you are killing it/reworking it soon :)